internal/daemon/worker/testing.go

```diff
@@ -552,13 +537,59 @@ func NewTestMultihopWorkers(t testing.TB,
 	childKmsWorker.w.conf.WorkerAuthStorageKms = nil
 
 	err = childKmsWorker.w.Start()
-	t.Cleanup(childKmsWorker.Shutdown)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	// Sleep so that workers can startup and connect.
-	time.Sleep(12 * time.Second)
+	// validate worker status reports to controller
+	wg := new(sync.WaitGroup)
+	wg.Add(4)
+	go func() {
+		defer wg.Done()
+		require.NoError(kmsWorker.Worker().WaitForNextSuccessfulStatusUpdate())
+		require.Eventually(func() bool {
+			_, ok := c.Controller().WorkerStatusUpdateTimes().Load(kmsWorker.Name())
+			return ok
+		}, 30*time.Second, 100*time.Millisecond)
+	}()
+	go func() {
+		defer wg.Done()
+		require.NoError(pkiWorker.Worker().WaitForNextSuccessfulStatusUpdate())
+		require.Eventually(func() bool {
+			_, ok := c.Controller().WorkerStatusUpdateTimes().Load(pkiWorker.Name())
+			return ok
+		}, 30*time.Second, 100*time.Millisecond)
+	}()
+	go func() {
+		defer wg.Done()
+		require.NoError(childPkiWorker.Worker().WaitForNextSuccessfulStatusUpdate())
+		require.Eventually(func() bool {
+			_, ok := c.Controller().WorkerStatusUpdateTimes().Load(childPkiWorker.Name())
+			return ok
+		}, 30*time.Second, 100*time.Millisecond)
+	}()
+	go func() {
+		defer wg.Done()
+		require.NoError(childKmsWorker.Worker().WaitForNextSuccessfulStatusUpdate())
+		require.Eventually(func() bool {
+			_, ok := c.Controller().WorkerStatusUpdateTimes().Load(childKmsWorker.Name())
+			return ok
+		}, 30*time.Second, 100*time.Millisecond)
+	}()
+	wg.Wait()
+
+	// validate connectivity between worker pool
+	require.Eventually(func() bool {
+		pkiPath, err := c.DownstreamWorkers().PathTo(c.Context(), pkiWorker.Worker().LastStatusSuccess().WorkerId)
+		if err != nil {
+			return false
+		}
+		childPkiPath, err := c.DownstreamWorkers().PathTo(c.Context(), childPkiWorker.Worker().LastStatusSuccess().WorkerId)
+		if err != nil {
```

Comment:
Maybe this should go in a testing_ent instead if we can't do this in community 🤔

Reply:
Should we move all the tests that reference this helper function into `_ent` files too?

Reply:
Yeah that'd be a requirement too.